manila/network/neutron/api.py

```diff
@@ -167,6 +182,15 @@ def create_port(self, tenant_id, network_id, host_id=None, subnet_id=None,
                 raise exception.PortLimitExceeded()
             raise exception.NetworkException(code=e.status_code,
                                              message=e.message)
+        except ks_exec.ConnectFailure:
+            LOG.warning('Create Port: Neutron connection failure')
+            # check if port is created in neutron else re-raise connectFailure
+            search_opts = {'device_id': device_id, 'network_id': network_id}
```

Comment:
Sadly this combination is not enough, we have `get_network_allocations_number()` (in the netapp driver this translates to the number of cluster nodes, usually 2 in our landscape) times ports with the same device_id + network_id combination.
Maybe we should start giving the ports different `name`s, just counting up the allocation numbers to have something unique for identification?
I.e. I imagine the caller of this method `def allocate_network()` to pass a name containing the `allocation_count`
WDYT?

Reply:
updated

Reply:
thanks